src/monio/AtlasWriter.h

```diff
@@ -35,64 +37,76 @@ class AtlasWriter {
-
+  /// \brief Creates all metadata and data from at Atlas field. For writing of field sets with no
+  ///        metadata.
+  void populateFileDataWithField(FileData& fileData,
+                           const atlas::Field& field,
+                           const std::string& writeName);
 
+ private:
+  /// \brief Called from the entry point with LFRic metadata.
   void populateMetadataWithField(Metadata& metadata,
                            const atlas::Field& field,
                            const consts::FieldMetadata& fieldMetadata,
                            const std::string& varName);
 
+  /// \brief Called from the entry point with no existing metadata.
   void populateMetadataWithField(Metadata& metadata,
                            const atlas::Field& field,
                            const std::string& varName);
 
+  /// \brief Called from the entry point with LFRic metadata.
   void populateDataWithField(Data& data,
                        const atlas::Field& field,
                        const std::vector<size_t>& lfricToAtlasMap,
                        const std::string& fieldName);
 
+  /// \brief Called from the entry point with no existing metadata.
   void populateDataWithField(Data& data,
                        const atlas::Field& field,
                        const std::vector<int> dimensions);
 
+  /// \brief With LFRic metadata, derives the container type and makes the call to populate it.
+  void populateDataContainerWithField(std::shared_ptr<monio::DataContainerBase>& dataContainer,
+                                const atlas::Field& field,
+                                const std::vector<size_t>& lfricToAtlasMap,
+                                const std::string& fieldName);
+
+  /// \brief Without metadata, derives the container type and makes the call to populate it.
+  void populateDataContainerWithField(std::shared_ptr<monio::DataContainerBase>& dataContainer,
+                                const atlas::Field& field,
+                                const std::vector<int>& dimensions);
+
+  /// \brief Iterates through field and populates vector with data from field in LFRic order.
   template<typename T> void populateDataVec(std::vector<T>& dataVec,
                                       const atlas::Field& field,
                                       const std::vector<size_t>& lfricToAtlasMap);
 
+  /// \brief Iterates through field and populates vector with data from field in Atlas order.
   template<typename T> void populateDataVec(std::vector<T>& dataVec,
                                       const atlas::Field& field,
                                       const std::vector<int>& dimensions);
 
+  /// \brief Returns a field formatted for writing to file.
   atlas::Field getWriteField(atlas::Field& inputField,
                        const std::string& writeName,
                        const bool noFirstLevel);
 
+  /// \brief Returns a field with a copy of the zeroth level of input field.
```

Comment:
Slightly confusing wording but I see why you have worded it like this. "a field with a copy of the 0th level" - so a 1D field?
Perhaps:
```suggestion
  /// \brief Returns a copy of the input field, with an additional level containing a copy of the zeroth level.
```
Or something to that effect.

Reply:
Cheers!